code/modules/mob/living/carbon/human/human.dm

```diff
@@ -835,36 +841,70 @@
 		return TRUE
 	return FALSE
 
-/mob/living/carbon/human/buckle_mob(mob/living/M, force = FALSE, check_loc = TRUE)
+/mob/living/carbon/human/proc/can_be_firemanned(mob/living/carbon/C)
+	if(ishuman(C) && !(C.mobility_flags & MOBILITY_STAND))
+		return TRUE
+	return FALSE
+
+/mob/living/carbon/human/proc/fireman_carry(mob/living/carbon/C)
+	if(can_be_firemanned(C))
+		visible_message("<span class='notice'>[src] starts lifting [C] onto their back...</span>", 
+			"<span class='notice'>You start lifting [C] onto your back...</span>")
+		if(do_after(src, 50, TRUE, C))
+			if(can_be_firemanned(C))//Second check to make sure they're still valid to be carried
+				if(!incapacitated(FALSE, TRUE))
+					buckle_mob(C, TRUE, TRUE, 90, 1, 0)
+					return
+		visible_message("<span class='warning'>[src] fails to fireman carry [C]!")

```

Comment:
this should be bound to an else I think

Reply:
it shouldn't, it is so that if any of those ifs fail it doesn't return and reaches that `visible_message`. I could turn them into one if but I felt like that would affect the readability.